frontend/src/Repos/WmsAllowedOperationRepo.ts

```diff
@@ -0,0 +1,116 @@
+import Cookies from 'js-cookie';
+import JsonApiRepo, { JsonApiMimeType, JsonApiResponse, QueryParams } from './JsonApiRepo';
+
+export interface WmsAllowedOperationCreate {
+  description: string;
+  allowedGroupIds: Array<string>;
+  allowedOperationIds: Array<string>;
+  securedLayerIds: Array<string>;
+}
+
+class WmsAllowedOperationRepo extends JsonApiRepo {
+
+  readonly wmsId: string;
+
+  constructor (wmsId: string) {
+    super('/api/v1/registry/wms/{parent_lookup_secured_service}/allowed-wms-operations/', 'Zugriffsregeln');
+    this.wmsId = wmsId;
+  }
+
+  async findAll (queryParams?: QueryParams): Promise<JsonApiResponse> {
+    const client = await JsonApiRepo.getClientInstance();
+    let jsonApiParams: any;
+    if (queryParams) {
+      jsonApiParams = {
+        'page[number]': queryParams.page,
+        'page[size]': queryParams.pageSize,
+        ...queryParams.filters
+      };
+      if (queryParams.ordering) {
+        jsonApiParams.sort = queryParams.ordering;
+      }
+    }
+    return await client['List' + this.resourcePath](this.wmsId, jsonApiParams);
+  }
+
+  async add (type: string, attributes: any, relationships?: any): Promise<JsonApiResponse> {
```

Comment:
ts is complaining about the type being used. In this case it needs to be specific. Can't you structure how the attributes and relationships for this case will look like?

Reply:
That's a good point.  Honestly, I feel unable to cope properly with the typing of JsonApi types in our project, because of the following problems:

- Our project team decided to try to work with the API in a generic fashion. Ideally, the frontend should adapt to new/changed attributes in types automatically. While we still evaluate this, I don't see that it makes sense to invest energy in typing the API object.
- There's flexibility in attributes and additional properties (included, relationships)

@Unraveler @jokiefer What do you think?